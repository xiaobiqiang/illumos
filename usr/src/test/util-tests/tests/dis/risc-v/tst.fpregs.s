/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of all of the fp register names.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:

libdifs_tefst:
	fadd.s	ft0, ft0, ft0
	fadd.s	ft1, ft1, ft1
	fadd.s	ft2, ft2, ft2
	fadd.s	ft3, ft3, ft3
	fadd.s	ft4, ft4, ft4
	fadd.s	ft5, ft5, ft5
	fadd.s	ft6, ft6, ft6
	fadd.s	ft7, ft7, ft7
	fadd.s	fs0, fs0, fs0
	fadd.s	fs1, fs1, fs1
	fadd.s	fa0, fa0, fa0
	fadd.s	fa1, fa1, fa1
	fadd.s	fa2, fa2, fa2
	fadd.s	fa3, fa3, fa3
	fadd.s	fa4, fa4, fa4
	fadd.s	fa5, fa5, fa5
	fadd.s	fa6, fa6, fa6
	fadd.s	fa7, fa7, fa7
	fadd.s	fs2, fs2, fs2
	fadd.s	fs3, fs3, fs3
	fadd.s	fs4, fs4, fs4
	fadd.s	fs5, fs5, fs5
	fadd.s	fs6, fs6, fs6
	fadd.s	fs7, fs7, fs7
	fadd.s	fs8, fs8, fs8
	fadd.s	fs9, fs9, fs9
	fadd.s	fs10, fs10, fs10
	fadd.s	fs11, fs11, fs11
	fadd.s	ft8, ft8, ft8
	fadd.s	ft9, ft9, ft9 
	fadd.s	ft10, ft10, ft10 
	fadd.s	ft11, ft11, ft11

	fadd.s	ft0, ft1, ft2
	fadd.s	ft1, ft2, ft3
	fadd.s	ft2, ft3, ft4
	fadd.s	ft3, ft4, ft5
	fadd.s	ft4, ft5, ft6
	fadd.s	ft5, ft6, ft7
	fadd.s	ft6, ft7, fs0
	fadd.s	ft7, fs0, fs1
	fadd.s	fs0, fs1, fa0
	fadd.s	fs1, fa0, fa1
	fadd.s	fa0, fa1, fa2
	fadd.s	fa1, fa2, fa3
	fadd.s	fa2, fa3, fa4
	fadd.s	fa3, fa4, fa5
	fadd.s	fa4, fa5, fa6
	fadd.s	fa5, fa6, fa7
	fadd.s	fa6, fa7, fs2
	fadd.s	fa7, fs2, fs3
	fadd.s	fs2, fs3, fs4
	fadd.s	fs3, fs4, fs5
	fadd.s	fs4, fs5, fs6
	fadd.s	fs5, fs6, fs7
	fadd.s	fs6, fs7, fs8
	fadd.s	fs7, fs8, fs9
	fadd.s	fs8, fs9, fs10
	fadd.s	fs9, fs10, fs11
	fadd.s	fs10, fs11, ft8
	fadd.s	fs11, ft8, ft9
	fadd.s	ft8, ft9, ft10
	fadd.s	ft9, ft10, ft11
	fadd.s	ft10, ft11, ft0
	fadd.s	ft11, ft0, ft1
.size libdis_test, [.-libdis_test]
