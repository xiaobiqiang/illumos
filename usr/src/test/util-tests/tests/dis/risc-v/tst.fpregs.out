    libdis_test:       53 70 00 00        fadd.s ft0,ft0,ft0
    libdis_test+0x4:   d3 f0 10 00        fadd.s ft1,ft1,ft1
    libdis_test+0x8:   53 71 21 00        fadd.s ft2,ft2,ft2
    libdis_test+0xc:   d3 f1 31 00        fadd.s ft3,ft3,ft3
    libdis_test+0x10:  53 72 42 00        fadd.s ft4,ft4,ft4
    libdis_test+0x14:  d3 f2 52 00        fadd.s ft5,ft5,ft5
    libdis_test+0x18:  53 73 63 00        fadd.s ft6,ft6,ft6
    libdis_test+0x1c:  d3 f3 73 00        fadd.s ft7,ft7,ft7
    libdis_test+0x20:  53 74 84 00        fadd.s fs0,fs0,fs0
    libdis_test+0x24:  d3 f4 94 00        fadd.s fs1,fs1,fs1
    libdis_test+0x28:  53 75 a5 00        fadd.s fa0,fa0,fa0
    libdis_test+0x2c:  d3 f5 b5 00        fadd.s fa1,fa1,fa1
    libdis_test+0x30:  53 76 c6 00        fadd.s fa2,fa2,fa2
    libdis_test+0x34:  d3 f6 d6 00        fadd.s fa3,fa3,fa3
    libdis_test+0x38:  53 77 e7 00        fadd.s fa4,fa4,fa4
    libdis_test+0x3c:  d3 f7 f7 00        fadd.s fa5,fa5,fa5
    libdis_test+0x40:  53 78 08 01        fadd.s fa6,fa6,fa6
    libdis_test+0x44:  d3 f8 18 01        fadd.s fa7,fa7,fa7
    libdis_test+0x48:  53 79 29 01        fadd.s fs2,fs2,fs2
    libdis_test+0x4c:  d3 f9 39 01        fadd.s fs3,fs3,fs3
    libdis_test+0x50:  53 7a 4a 01        fadd.s fs4,fs4,fs4
    libdis_test+0x54:  d3 fa 5a 01        fadd.s fs5,fs5,fs5
    libdis_test+0x58:  53 7b 6b 01        fadd.s fs6,fs6,fs6
    libdis_test+0x5c:  d3 fb 7b 01        fadd.s fs7,fs7,fs7
    libdis_test+0x60:  53 7c 8c 01        fadd.s fs8,fs8,fs8
    libdis_test+0x64:  d3 fc 9c 01        fadd.s fs9,fs9,fs9
    libdis_test+0x68:  53 7d ad 01        fadd.s fs10,fs10,fs10
    libdis_test+0x6c:  d3 fd bd 01        fadd.s fs11,fs11,fs11
    libdis_test+0x70:  53 7e ce 01        fadd.s ft8,ft8,ft8
    libdis_test+0x74:  d3 fe de 01        fadd.s ft9,ft9,ft9
    libdis_test+0x78:  53 7f ef 01        fadd.s ft10,ft10,ft10
    libdis_test+0x7c:  d3 ff ff 01        fadd.s ft11,ft11,ft11
    libdis_test+0x80:  53 f0 20 00        fadd.s ft0,ft1,ft2
    libdis_test+0x84:  d3 70 31 00        fadd.s ft1,ft2,ft3
    libdis_test+0x88:  53 f1 41 00        fadd.s ft2,ft3,ft4
    libdis_test+0x8c:  d3 71 52 00        fadd.s ft3,ft4,ft5
    libdis_test+0x90:  53 f2 62 00        fadd.s ft4,ft5,ft6
    libdis_test+0x94:  d3 72 73 00        fadd.s ft5,ft6,ft7
    libdis_test+0x98:  53 f3 83 00        fadd.s ft6,ft7,fs0
    libdis_test+0x9c:  d3 73 94 00        fadd.s ft7,fs0,fs1
    libdis_test+0xa0:  53 f4 a4 00        fadd.s fs0,fs1,fa0
    libdis_test+0xa4:  d3 74 b5 00        fadd.s fs1,fa0,fa1
    libdis_test+0xa8:  53 f5 c5 00        fadd.s fa0,fa1,fa2
    libdis_test+0xac:  d3 75 d6 00        fadd.s fa1,fa2,fa3
    libdis_test+0xb0:  53 f6 e6 00        fadd.s fa2,fa3,fa4
    libdis_test+0xb4:  d3 76 f7 00        fadd.s fa3,fa4,fa5
    libdis_test+0xb8:  53 f7 07 01        fadd.s fa4,fa5,fa6
    libdis_test+0xbc:  d3 77 18 01        fadd.s fa5,fa6,fa7
    libdis_test+0xc0:  53 f8 28 01        fadd.s fa6,fa7,fs2
    libdis_test+0xc4:  d3 78 39 01        fadd.s fa7,fs2,fs3
    libdis_test+0xc8:  53 f9 49 01        fadd.s fs2,fs3,fs4
    libdis_test+0xcc:  d3 79 5a 01        fadd.s fs3,fs4,fs5
    libdis_test+0xd0:  53 fa 6a 01        fadd.s fs4,fs5,fs6
    libdis_test+0xd4:  d3 7a 7b 01        fadd.s fs5,fs6,fs7
    libdis_test+0xd8:  53 fb 8b 01        fadd.s fs6,fs7,fs8
    libdis_test+0xdc:  d3 7b 9c 01        fadd.s fs7,fs8,fs9
    libdis_test+0xe0:  53 fc ac 01        fadd.s fs8,fs9,fs10
    libdis_test+0xe4:  d3 7c bd 01        fadd.s fs9,fs10,fs11
    libdis_test+0xe8:  53 fd cd 01        fadd.s fs10,fs11,ft8
    libdis_test+0xec:  d3 7d de 01        fadd.s fs11,ft8,ft9
    libdis_test+0xf0:  53 fe ee 01        fadd.s ft8,ft9,ft10
    libdis_test+0xf4:  d3 7e ff 01        fadd.s ft9,ft10,ft11
    libdis_test+0xf8:  53 ff 0f 00        fadd.s ft10,ft11,ft0
    libdis_test+0xfc:  d3 7f 10 00        fadd.s ft11,ft0,ft1
